### PR TITLE
Set last puck position on every settings change.

### PR DESCRIPTION
**Pull request by yunikkk**
Set last animation position before adding location layer to the style to avoid displaying puck at (0,0) position.

Currently any property that is changed in runtime via [LocationComponentSettingsInterface.kt](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk-base/src/main/java/com/mapbox/maps/plugin/locationcomponent/generated/LocationComponentSettingsInterface.kt) will shortly reset puck to the (0,0) coordinates, while new location is not retrieved from the LocationProvider. 
Can be reproduced in the test app, changing the puck style : 

| Before | After | 
| ----- | -----|
| <video src='https://user-images.githubusercontent.com/2265924/141315463-694d7d10-ea7f-4f75-8d0c-cd4ffde34b5f.mp4' width=250/>  | <video src='https://user-images.githubusercontent.com/2265924/141317815-400c659a-764e-46f0-8c5b-422037bb84be.mp4' width=250/>|

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Fix puck jumps to nullisland when location plugin settings are changed.</changelog>`.

### CODE REVIEW

**Review by kiryldz — CHANGES_REQUESTED** [review]
@yunikkk congrats on first contribution, several points:
 - sign CLA
 - update PR description per template
 - video was not uploaded in PR description for some reason
 - missing tests for fixes
 - rebase your PR on latest `main`, for some reason code gen job does fail, may be not connected but to be sure

**Comment by pengdev** [comment]
Looks good! please also attach a before/after video/gif in the PR description and add tests to cover the change.

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt [hunk 1]
```diff
@@ -109,12 +108,12 @@ internal class LocationPuckManager(
   }
 
   fun updateCurrentPosition(vararg points: Point, options: (ValueAnimator.() -> Unit)? = null) {
-    val targets = if (lastLocation == null) arrayOf(*points, *points) else {
+    val targets = lastLocation?.let {
       show()
-      arrayOf(lastLocation, *points)
-    }
+      arrayOf(it, *points)
+    } ?: arrayOf(*points, *points)
```
Comment (kiryldz) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:114]: is this actually part of the fix as well?

Comment (yunikkk) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:114]: No, actually. Just removed unsafe cast that could possibly break if concurrency would take place

**Review by tobrun — APPROVED** [review]
:shipit:

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by pengdev — APPROVED** [review]
🚢

File: plugin-locationcomponent/src/test/java/com/mapbox/maps/util/MockKExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.mapbox.maps.util
+
+import io.mockk.MockKMatcherScope
+
+
```
Comment (kiryldz) [inline plugin-locationcomponent/src/test/java/com/mapbox/maps/util/MockKExtensions.kt:5]: nit: remove extra line

Comment (yunikkk) [reply plugin-locationcomponent/src/test/java/com/mapbox/maps/util/MockKExtensions.kt:5]: It was a line for the comment!

File: plugin-locationcomponent/src/test/java/com/mapbox/maps/util/MockKExtensions.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.mapbox.maps.util
+
+import io.mockk.MockKMatcherScope
+
+// vararg capture isn't implemented, here's a workaround to collect it https://github.com/mockk/mockk/issues/432
+inline fun <reified T : Any> MockKMatcherScope.captureVararg(
```
Comment (yunikkk) [inline plugin-locationcomponent/src/test/java/com/mapbox/maps/util/MockKExtensions.kt:6]: @mapbox/maps-android added helper to capture varargs

**Comment by ank27** [comment]
lint is failing, just check that. else looks good 🚢

**Review by ank27 — APPROVED** [review]

**Pull request merged**